server/src/main/java/org/elasticsearch/common/logging/HeaderWarning.java

```diff
@@ -408,6 +409,17 @@ public static void addWarning(String message, Object... params) {
         addWarning(THREAD_CONTEXT, message, params);
     }
 
+    /**
+     * Adds a warning header without any formatting or prefix
+     */
+    public static void addRawWarning(String warningHeader) {
+        addRawWarning(THREAD_CONTEXT, warningHeader);
+    }
+
+    public static Set<String> getWarnings() {
+        return getWarnings(THREAD_CONTEXT);
+    }
```

Comment:
How did we manage to do this shift without this before? I feel like we shift threads and merge warnings in other ways without requiring this.

Reply:
I guess rather than make a change to the core warnings stuff I'd prefer we continue to hack around it until we can replace the thread-local-warnings in ESQL. With, like, something else.

Reply:
Changed to what Enrich does (A custom class that does a similar thing, but with all the headers). And moved the logic instead to AsyncOperator, as to avoid repeating it, as it looks quite a standard thing to do there.
Added it to the javadoc too